src/mmda/recipes/recipe.py

```diff
@@ -0,0 +1,62 @@
+"""
+
+
+@kylel
+
+"""
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+from mmda.types import *
+from mmda.parsers.pdfplumber_parser import PDFPlumberParser
+from mmda.rasterizers.rasterizer import PDF2ImageRasterizer
+from mmda.predictors.heuristic_predictors.dictionary_word_predictor import DictionaryWordPredictor
+from mmda.predictors.lp_predictors import LayoutParserPredictor
+from mmda.predictors.hf_predictors.token_classification_predictor import \
+    IVILATokenClassificationPredictor
+
+
+class Recipe:
+    def run(self, doc: Document) -> Document:
+        raise NotImplementedError
+
+
+class CoreRecipe(Recipe):
+    def __init__(self,
+                 effdet_publaynet_predictor_path: str = 'lp://efficientdet/PubLayNet',
+                 effdet_mfd_predictor_path: str = 'lp://efficientdet/MFD',
+                 vila_predictor_path: str = 'allenai/ivila-row-layoutlm-finetuned-s2vl-v2'):
+        logger.info("Instantiating recipe...")
+        self.parser = PDFPlumberParser()
+        self.rasterizer = PDF2ImageRasterizer()
```

Comment:
this is just for my understanding, but I don't think we explicitly run a rasterizer in the SPP pipeline, we just run PDFPlumber and LayoutParser. Can you clarify why it is run here and not there?

Reply:
Hmm, interesting. How is SPP obtaining images of the pages of the PDF? LayoutParser operates over those images

Reply:
I feel like this must've been run somehow? LayoutParser operates over the `document.images` which `PDFPlumberParser` doesnt produce;  it's only producable with `PDF2ImageRasterizer`.